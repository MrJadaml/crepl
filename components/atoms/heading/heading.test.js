import React from 'react';
import { Heading } from './index';
import { cleanup, render } from '@testing-library/react';
import { getByText } from '@testing-library/dom';

afterEach(cleanup);

describe('Heading', () => {
  it('should render an h1 heading by default', () => {
    const { asFragment } = render(<Heading>Hi Bob</Heading>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a dynamic heading', () => {
    const { asFragment } = render(<Heading type="h2">Hi Carl</Heading>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children', () => {
    const { getByText } = render(<Heading type="h6">Hi Foo</Heading>);
    getByText('Hi Foo');
  });
});
