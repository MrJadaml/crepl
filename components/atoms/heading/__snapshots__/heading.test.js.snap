// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render an h1 heading 1`] = `
<DocumentFragment>
  <h1
    class="h1"
  >
    Hi Bob
  </h1>
</DocumentFragment>
`;

exports[`Heading should render an h2 heading 1`] = `
<DocumentFragment>
  <h2
    class="h2"
  >
    Hi Carl
  </h2>
</DocumentFragment>
`;
